### PR TITLE
OIA-55: Implement collector config extensions

### PR DESCRIPTION
**Pull request by fooker**
### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/OIA-55

### CODE REVIEW

**Review by RangerRick — CHANGES_REQUESTED** [review]
this looks good to me, but it might be nice to see a test to make sure reload behaves properly after things change on both sides (local and OIA configs)

seems like daemon reloads not working right is a regular issue with support stuff, best to have an explicit test

**Comment by fooker** [comment]
There are forced smoke tests for this: https://app.circleci.com/pipelines/github/OpenNMS/opennms?branch=jira%2FOIA-55-smoke

**Comment by fooker** [comment]
> this looks good to me, but it might be nice to see a test to make sure reload behaves properly after things change on both sides (local and OIA configs)
> 
> seems like daemon reloads not working right is a regular issue with support stuff, best to have an explicit test

Done.

File: opennms-services/src/test/java/org/opennms/netmgt/collectd/CollectdExternalIT.java [hunk 1]
```diff
@@ -0,0 +1,307 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2007-2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
```
Comment (christianpape) [inline opennms-services/src/test/java/org/opennms/netmgt/collectd/CollectdExternalIT.java:5]:
```suggestion
 * Copyright (C) 2007-2023 The OpenNMS Group, Inc.
 * OpenNMS(R) is Copyright (C) 1999-2023 The OpenNMS Group, Inc.
```

**Review by christianpape — APPROVED** [review]
LGTM.

**Review by RangerRick — APPROVED** [review]

**Review by RangerRick — APPROVED** [review]

**Pull request merged**